metricshub-web/react/src/components/config/Editor/ConfigEditorContainer.jsx

```diff
@@ -41,36 +42,110 @@ export default function ConfigEditorContainer() {
-	 */
+	// onChange handler for editor
 	const onChange = (v) => {
 		setLocal(v);
 		debouncedPush(v);
 	};
 
-	/**
-	 * Handle save action from the editor.
-	 * Dispatches save action to Redux store.
-	 * No-op if no file is selected or already saving.
-	 * @type {()=>void}
-	 */
-	const onSave = () => {
-		if (!selected || saving) return;
-		dispatch(saveConfig({ name: selected, content: local, skipValidation: false }));
-	};
+	// dialog state for showing validation errors on save
+	const [dialog, setDialog] = React.useState(null); // { open, message, from?, to? }
+
+	// editor view ref - set by YamlEditor via onEditorReady
+	const editorViewRef = React.useRef(null);
+
+	// validate-then-save: run validation first, then either save or open dialog listing errors
+	// dialog: null = closed; otherwise { open:true, message, from?, to? }
+	const wrappedSave = React.useCallback(async () => {
+		if (!canSave) return;
+		try {
+			const res = await dispatch(validateConfig({ name: selected, content: local })).unwrap();
+			const result = res?.result ?? { valid: true };
+
+			// if invalid -> extract first error, compute range via util, and show dialog
+			if (result?.valid === false) {
+				const first = Array.isArray(result.errors)
+					? result.errors[0]
+					: result.error
+						? { message: result.error }
+						: null;
+
+				if (first) {
+					const { from, to, message } = extractYamlErrorRange(first, editorViewRef.current);
+					setDialog({
+						open: true,
+						message: shortYamlError(message || "Validation failed"),
+						from,
+						to,
+					});
+					return;
+				}
+
+				setDialog({ open: true, message: "Validation failed" });
+				return;
+			}
+
+			// valid → save
+			dispatch(saveConfig({ name: selected, content: local, skipValidation: false }));
+		} catch (e) {
+			setDialog({ open: true, message: e?.message || "Validation failed" });
+		}
+	}, [canSave, dispatch, selected, local]);
+
+	// expose wrappedSave to parent via ref
+	React.useImperativeHandle(
+		ref,
+		() => ({
+			save: wrappedSave,
+		}),
+		[wrappedSave],
+	);
+
+	// validation function for editor linting
+	const validateFn = React.useCallback(
+		async (content, name) => {
+			try {
+				const res = await dispatch(validateConfig({ name, content })).unwrap();
+				return res?.result ?? { valid: true };
+			} catch {
+				return { valid: true }; // fallback to no lint markers
+			}
+		},
+		[dispatch],
+	);
 
 	return (
-		<ConfigEditor
-			value={local}
-			readOnly={!selected}
-			onChange={onChange}
-			onSave={onSave}
-			height="100%"
-		/>
+		<>
+			<ConfigEditor
+				value={local}
+				readOnly={!selected}
+				onChange={onChange}
+				onSave={wrappedSave}
+				canSave={canSave}
+				height="100%"
+				fileName={selected}
+				validateFn={validateFn}
+			/>
+			<QuestionDialog
+				open={!!dialog?.open}
+				title="Please fix this error"
+				question={
+					<>
+						<p style={{ whiteSpace: "pre-wrap", margin: 0 }}>{dialog?.message}</p>
```

Comment:
Use `<Typography>`  in stead of `<p>` with sx. to apply consistent typography styles (font, size, line height, color, etc.) from your MUI theme.